pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -734,13 +734,31 @@ private void writeMetadata()
               DateTimeFormatSpec formatSpec = _config.getDateTimeFormatSpec();
               Preconditions.checkNotNull(formatSpec, "DateTimeFormatSpec must exist for SimpleDate");
               DateTimeFormatter dateTimeFormatter = formatSpec.getDateTimeFormatter();
-              startTime = dateTimeFormatter.parseMillis(startTimeStr);
-              endTime = dateTimeFormatter.parseMillis(endTimeStr);
+              try {
+                startTime = dateTimeFormatter.parseMillis(startTimeStr);
+                endTime = dateTimeFormatter.parseMillis(endTimeStr);
+              } catch (Exception e) {
+                if (_config.isContinueOnError()) {
+                  startTime = Long.MIN_VALUE;
```

Comment:
Setting min/max here can cause problem later. When encountering exception, we should always use -1 as start/end time and MILLIS as time unit to be consistent with the handling in the segment ZK metadata

Reply:
These are just placeholder values. In the lines below, since the start/end is invalid, we won't be setting these in final metadata object.